pkg/controller/service/service_controller.go

```diff
@@ -660,30 +673,36 @@ func getNodeConditionPredicate() corelisters.NodeConditionPredicate {
 // nodeSyncLoop handles updating the hosts pointed to by all load
 // balancers whenever the set of nodes in the cluster changes.
 func (s *ServiceController) nodeSyncLoop() {
-	newHosts, err := s.nodeLister.ListWithPredicate(getNodeConditionPredicate())
-	if err != nil {
-		runtime.HandleError(fmt.Errorf("Failed to retrieve current set of nodes from node lister: %v", err))
-		return
+	allNewHosts := make([]*v1.Node, 0)
+	for _, nodeLister := range s.nodeListers {
+		newHosts, err := nodeLister.ListWithPredicate(getNodeConditionPredicate())
+		if err != nil {
+			// TODO - check error for RP not available, skip
+			runtime.HandleError(fmt.Errorf("Failed to retrieve current set of nodes from node lister: %v", err))
```

Comment:
nit: if it can hint out the culprit node lister/RP, the logging would be more useful.

Reply:
> nit: if it can hint out the culprit node lister/RP, the logging would be more useful.

NodeLister is internally generated, it does not talk to RP directly so error will be internal only.